internal/registry/registry_test.go

```diff
@@ -400,4 +467,8 @@ func TestPollProvider(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	err = dstore.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Let's add a simple migration test too. Especially since this is a more ad-hoc migration strategy.